### PR TITLE
GnuPg Extended Key Format; code analysis workflows

### LINKED ISSUE #16: Add Coverity scan.
As sexp sources are used by rnp, we should make sure that it doesn't have Coverity alerts.

### LINKED ISSUE #17: Add Codecov upload/scan.
...to ensure tests coverage.

### PR DESCRIPTION
**Pull request by maxirmx**

### CODE REVIEW

**Comment by maxirmx** [comment]
This PR implements:

- Codecov, coverity, CodeQL workflows and related fixes including 99% test coverage of the library
(@ronaldtse  Please move codecov and  coverity  to projects setting, thank you)
- Support for GnuPG extended key format syntax. Although it is something outside of pure SEXP scope, I believe its better to put it here for the sake of future maintainability.

File: include/sexp/sexp-error.h [hunk 1]
```diff
@@ -40,25 +40,31 @@ class sexp_exception_t : public std::exception {
     enum severity { error = 0, warning = 1 };
 
   protected:
-    static severity verbosity_;
-    static bool     interactive_;
+    static severity vrb;
```
Comment (ni4) [inline include/sexp/sexp-error.h]: Do we really need to make these renames? It's more readable with full names with underscore attached, as for me.

File: src/ext-key-format.cpp [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+// true if allowed in the name field
+bool ext_key_input_stream_t::namechar[256];
+
+/*
+ * ext_key_input_stream_t::initializeCharacterTables
+ */
+ext_key_input_stream_t::ext_key_input_stream_t(std::istream *i, size_t md)
+    : sexp_input_stream_t(i, md), is_scanning_value(false), has_key(false)
+{
+    if (!initialized) {
```
Comment (ni4) [inline src/ext-key-format.cpp]: Why not to store just static pre-set const table with 0/1?

**Review by ni4 — COMMENTED** [review]
Overall LGTM, just few minor improvements could be made.

File: include/sexp/sexp-error.h (hunk 1 above)
Comment (maxirmx) [reply include/sexp/sexp-error.h]: I did full names using but using other approach. Sorry, underscopes at the end harm my aestetic senses

Comment (ni4) [reply include/sexp/sexp-error.h]: @maxirmx This is actually recommended by Google C++ style guide: https://google.github.io/styleguide/cppguide.html#Variable_Names
Anyway, it's up to you.

**Review by ni4 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**